docs/DeveloperGuide.md

```diff
@@ -175,7 +219,20 @@ The second sequence diagram given below shows the detailed interaction that aces
+<small><i>Figure 12</i></small> 
+
+### Implementation Considerations For Circuit Commands (Dash) <a name="impl-cons-dash"></a>
+This section describes the methods taken into consideration while implementing the Circuit Commands.
+
+#### Rationale Behind The Implementation of Circuit Template <a name="rationale-circuit-dash"></a>
+This section details the rationale for implimenting the circuit template as described in the prior sections. Selecting a way to build circuits is the primary purpose of the system and hence a way to design the circuit is required. Since the system is designed primarily as a command line interface, creating complex customizable circuits using a graphical interface is beyond the scope of this system. To implement a circuit four different `template` classes are used, which inherit from a superclass `template`. The four different templates are also presented as simple logos to show how the templates would look like as real life circuits. Furthermore, the function to add components to the circuit calculates the effective value with the components that are already present. For example when a resistor is added in parallel, the effective resistance of the aforementioned resistor is added to the effective resistance of the resistor already present in the circuit template. The two resistors are then replaced by one resistor with the same effective resistance. Similarly, other components are also added and abstracted into a single component of the matching component type.
+
+#### Alternatives Considered <a name="alt-circuit-dash"></a>
+This section illustrates the other alternatives considered for implementing the Circuit Template.
+
+* Linked List: A linked list implementation involves using each circuit component as a node and the connections between the nodes as wires. The head of the linked list is the voltage source and the a loop is present from the tail node number 2 (i.e index 1). However, such an approach allows users to add any component even when certain circuits are not usually built with those components. To fulfill the purpose of the system to cater to the needs of CG111 students this approach is avoided. The circuit templates on the other hand allow the user to distiguish between different circuits and do so effectively.     
+
+* Adjacency Matrix: The Adjacency matrix graph data structure is also considered as it allows connections between nodes. Each node is a component and a matrix is derived where a 1 represents a connection between the row of that element and the column of the same element. This approach is rejected due to high time complexity. For example, adding a component to the matrix requires O(V^2) time complexity where V is the number of vertices. Similarly, removing a component from the matrix also requires the same time complexity. On the other hand, the chosen implementation only requires O(1) time complexity for the same functions, leading to a more efficient and faster system.π
```

Comment:
Lol the random π at the end

Reply:
Also, since we talk about adjacency matrix here, consider calling the implementation "Graph with Adj Matrix" or something since an adj matrix is a way to rep a graph.

Also will have to add the term Adjacency Matrix to the glossary.

Reply:
Adding an edge to an AM is O(1), checking if an edge exists is also O(1). Removing an edge is also O(1). The issue with the AM is the high space complexity.